### PR TITLE
actually preventing duplicate nations for real this time

### PR DESCRIPTION
**Pull request by nsupc**
- fixed queue update logic to prevent any duplicate entries
- templates now stored without %/%25
- it is added directly to the telegram url instead

### CODE REVIEW

**Comment by nsupc** [comment]
the previous issue was that queue.update only checked and removed the names of previously recruited nations from the list before adding new items to the queue, meaning that if the queue wasn't entirely emptied, duplicate records for previously unrecruited nations would be introduced. oops...

**Comment by brrscc** [comment]
Nice catch! Looks good to me.

File: components/recruitment.py [hunk 1]
```diff
@@ -10,6 +10,11 @@
 def get_recruit_embed(user_id: int, bot: RecruitBot) -> Optional[Tuple[discord.Embed, RecruitView]]:
     user = bot.rusers.get(user_id)
 
+    # this will never happen because we validate that the user exists in the command,
+    # but my IDE is complaining without it
+    if not user:
+        return None
```
Comment (Darcness) [inline components/recruitment.py:16]: Don't return None, that'll freak out the IDE more.  Just raise some BS exception.

**Review by Darcness — APPROVED** [review]

File: components/recruitment.py (hunk 1 above)
Comment (nsupc) [reply components/recruitment.py:16]: Yeah we could change that in future, I think this is functional for now though

**Pull request merged**